src/Core/File/Exception/FileException.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\File\Exception;
+
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Base exception for files
+ *
+ * @todo: there are very similar exceptions in Core/Image. Images are also files, should/how could these be unified? (it's a mess)
```

Comment:
Should `ImageException` extend `FileException` ?

I'm not sure both are equivalent because there can be many image exceptions not related to purely the file (invalid format, conversion error, ...) I guess we could have an `ImageFileException` but then it couldn't extend both `ImageException` and `FileException` so it wouldn't work

Or we can just remove the image exceptions directly related to files (like `ImageNotFoundException` for example) and use only `FileException` related exceptions in these cases

Reply:
lets keep it as it is then